// @nebular/theme/esm5/components/cdk/a11y/descendant-key-manager.js
var __extends = (this && this.__extends) || (function () {
    var extendStatics = function (d, b) {
        extendStatics = Object.setPrototypeOf ||
            ({ __proto__: [] } instanceof Array && function (d, b) { d.__proto__ = b; }) ||
            function (d, b) { for (var p in b) if (b.hasOwnProperty(p)) d[p] = b[p]; };
        return extendStatics(d, b);
    };
    return function (d, b) {
        extendStatics(d, b);
        function __() { this.constructor = d; }
        d.prototype = b === null ? Object.create(b) : (__.prototype = b.prototype, new __());
    };
})();
import { ActiveDescendantKeyManager } from '@angular/cdk/a11y';
var NbActiveDescendantKeyManager = /** @class */ (function (_super) {
    __extends(NbActiveDescendantKeyManager, _super);
    function NbActiveDescendantKeyManager() {
        return _super !== null && _super.apply(this, arguments) || this;
    }
    return NbActiveDescendantKeyManager;
}(ActiveDescendantKeyManager));
export { NbActiveDescendantKeyManager };
var NbActiveDescendantKeyManagerFactoryService = /** @class */ (function () {
    function NbActiveDescendantKeyManagerFactoryService() {
    }
    NbActiveDescendantKeyManagerFactoryService.prototype.create = function (items) {
        return new NbActiveDescendantKeyManager(items);
    };
    return NbActiveDescendantKeyManagerFactoryService;
}());
export { NbActiveDescendantKeyManagerFactoryService };
export var NbKeyManagerActiveItemMode;
(function (NbKeyManagerActiveItemMode) {
    NbKeyManagerActiveItemMode[NbKeyManagerActiveItemMode["RESET_ACTIVE"] = -1] = "RESET_ACTIVE";
    NbKeyManagerActiveItemMode[NbKeyManagerActiveItemMode["FIRST_ACTIVE"] = 0] = "FIRST_ACTIVE";
})(NbKeyManagerActiveItemMode || (NbKeyManagerActiveItemMode = {}));
